planner/core/task.go

```diff
@@ -1061,28 +1066,64 @@ func (p *PhysicalTopN) pushTopNDownToDynamicPartition(copTsk *copTask) (task, bo
 		tblScan = finalTblScanPlan.(*PhysicalTableScan)
 		tblInfo = tblScan.Table
 	}
+	if len(copTsk.idxMergePartPlans) > 0 {
+		partialFinalScan = make([]PhysicalPlan, 0, len(copTsk.idxMergePartPlans))
+		for _, scan := range copTsk.idxMergePartPlans {
+			clonedScan, err := scan.Clone()
+			if err != nil {
+				return nil, false
+			}
+			partialClonedScan = append(partialClonedScan, clonedScan)
+			finalScan := clonedScan
+			for len(finalScan.Children()) > 0 {
+				finalScan = finalScan.Children()[0]
```

Comment:
what‘s this mean (collect its children)
or: what's diff between partialClonedScan and partialFinalScan here

Reply:
It's a sad story that.... you can only get the *model.TableInfo from `IndexScan` itself. so we need find the leaf of the `partialFinalScan `.

In other words, `partialFinalScan` might be `Selection->Scan` so you cannot get *model.TableInfo from the top Selection.

Reply:
1. I think the naming is confusing. Sometimes we describe an `IndexMerge` as consisting of several partial paths and a final path. Each path contains only one `Scan`. You can just call it `partialScan`.
2. It looks like we don't need a `Clone()` actually.

Reply:
Don't forget to have a look at this comment.

Reply:
+1, seem a little tricky here...